### DISCUSSION TITLE
Creating Dataset & Splitting

### DISCUSSION THREAD

**Discussion by SamWright822** [pyg-team/pytorch_geometric#4190] [Q&A] [answered]
Hi! 
I am trying to create a torch geometric dataset, and I'm getting some errors (and am a bit confused as to what I'm doing). I have 2047 sets of data (where my data is shaped 12,3). I am trying to make the dataset by using the following code: 

```
import torch_geometric
from torch_geometric.data import Data, Dataset

feat = torch.from_numpy(features) #shape (2047,12,3)
ys = torch.from_numpy(labels) #shape (2047,12,3)

traj_data = torch_geometric.data.Data(x = feat.to(torch.float32), y = ys.to(torch.float32))

```

But I think that I am doing this incorrectly, because if I try to split my data, I get a weird error. Here is what I do:

```
training_split = 1200
train_loader = torch_geometric.data.DataLoader(traj_data[:training_split], batch_size = 1, shuffle = False)

```

Which gives me an error. I do not need to do anything special with the dataset, I'm just trying to use it as an input to my model, which requires torch geometric datasets. 

I'd appreciate any help, though this might be a silly question. I am very new to this :)

Thanks!

**Comment by rusty1s** [comment, marked as answer]
Note that `DataLoader` expects a list of `data` objects:
```python
train_data_list = [Data(x=x[i], y=y[i]) for i range(feat.size(0))]
loader = DataLoader(train_data_list, batch_size=..., ...)
```
